File: practicas/practica_2/Deepspace_Ruby_P2/lib/DamageToUI.rb
#encoding:utf-8

# File: DamageToUI.rb
# Author: Profesor

module Deepspace

# Brief: Representación ToUI de Damage
class DamageToUI
  
  # Consultores
  attr_reader :nWeapons, :weapons, :nShields
  

  # Brief: Constructor con un parámetro
  # Param d: Instancia de la clase Damage
  def initialize (d) 
    @nWeapons=d.nWeapons
    @nShields=d.nShields
    tmp=d.weapons
    if (tmp!=nil) then
      @weapons=Array.new(tmp)
    else
      @weapons=nil
    end
  end
    
  public
  
  # Brief: Devuelve la información de la colección de armas concretas 
  #        a eliminar
  # Return: String con la información de la colección de armas concretas 
  #         a eliminar
  def getWeaponInfo() 
    out = "";
    
    if (nWeapons == -1) then
      out += "[" + @weapons.join(", ") + "]"
    else
      out += @nWeapons.to_s
    end
    return out
  end
  
  # Brief: Función para representar una instancia de la clase DamageToUI
  #        en un string
  # Return: out String que representa una instancia de la clase DamageToUI
  def to_s
    out = "Weapons: " + getWeaponInfo() + ", Shields: #{@nShields}"
  end
  
end # Class

end # Module

File: practicas/practica_2/deepspace_Java_P2/src/main/java/deepspace/DamageToUI.java
/**
 * @file DamageToUI.java
 * @author Profesor
 */
package deepspace;

import java.util.ArrayList;

/**
 * @brief Representación ToUI de Damage
 */
public class DamageToUI {
    private int nWeapons;
    private ArrayList<WeaponType> weapons;
    private int nShields;

    /**
     * @brief Constructor con un parámetro
     * @param d Instancia de la clase Damage
     */
    DamageToUI(Damage d) {
        nWeapons=d.getNWeapons();
        nShields=d.getNShields();
        ArrayList<WeaponType> tmp=d.getWeapons();
        if (tmp!=null) {
            weapons=new ArrayList<WeaponType>(tmp);
        }
        else
            weapons=null;
    }
    
    /**
     * @brief Consultor del número de armas a eliminar
     * @return Numero de Weapons
     */
    public int getNWeapons() {
        return nWeapons;
    }
    
    /**
     * @breif Consultor de la colección de armas concretas a eliminar
     * @return ArrayList weapons
     */
    public ArrayList<WeaponType> getWeapons() {
        return weapons;
    }

    /**
     * @brief Consultor del número de potenciadores de escudo a eliminar
     * @return nShields
     */
    public int getNShields() {
        return nShields;
    }
    
    /**
     * @brief Devuelve la información de la colección de armas concretas 
     *        a eliminar
     * @return String con la información de la colección de armas concretas 
     *         a eliminar
     */
    public String getWeaponInfo() {
      String out = "";
      
      if (nWeapons == -1) {
          out += weapons.toString();
      } else {
          out += nWeapons;
      }
      return out;
    }
}